todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModelTest.java

```diff
@@ -0,0 +1,143 @@
+
+import android.support.annotation.StringRes;
+
+import com.example.android.architecture.blueprints.todoapp.R;
+import com.example.android.architecture.blueprints.todoapp.data.Task;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+import com.example.android.architecture.blueprints.todoapp.util.providers.BaseResourceProvider;
+import com.google.common.collect.Lists;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.List;
+
+import rx.Observable;
+import rx.observers.TestSubscriber;
+
+import static org.hamcrest.CoreMatchers.any;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for the implementation of {@link StatisticsViewModel}
+ */
+public class StatisticsViewModelTest {
+
+    private static final String NO_TASKS = "no tasks";
+    private static final String ACTIVE_COMPLETED_TASKS = "Active tasks: %1$d \n Completed tasks: %2$d";
+
+    private List<Task> mTasks;
+
+    @Mock
+    private TasksRepository mTasksRepository;
+
+    @Mock
+    private BaseResourceProvider mResourceProvider;
+
+    private StatisticsViewModel mViewModel;
+
+    private TestSubscriber<Boolean> mProgressIndicatorTestSubscriber;
+
+    private TestSubscriber<String> mStatisticsTestSubscriber;
+
+    @Before
+    public void setupStatisticsPresenter() {
+        // Mockito has a very convenient way to inject mocks by using the @Mock annotation. To
+        // inject the mocks in the test the initMocks method needs to be called.
+        MockitoAnnotations.initMocks(this);
+
+        // Get a reference to the class under test
+        mViewModel = new StatisticsViewModel(mTasksRepository, mResourceProvider);
+
+        // We subscribe the tasks to 3, with one active and two completed
+        mTasks = Lists.newArrayList(
+                new Task("Title1", "Description1"),
+                new Task("Title2", "Description2", true),
+                new Task("Title3", "Description3", true));
+        mProgressIndicatorTestSubscriber = new TestSubscriber<>();
+        mStatisticsTestSubscriber = new TestSubscriber<>();
+    }
+
+    @Test
+    public void getProgressIndicator_emitsFalseInitially() {
+        // When subscribing to the progress indicator
+        mViewModel.getProgressIndicator().subscribe(mProgressIndicatorTestSubscriber);
+
+        // One value: false, is emitted
+        mProgressIndicatorTestSubscriber.assertValue(false);
+    }
+
+    @Test
+    public void getStatistics_withTasks_returnsCorrectData() {
+        //Given a list of tasks in the repository
+        when(mTasksRepository.getTasks()).thenReturn(Observable.just(mTasks));
+        // And string resources
+        withFormattedText(R.string.statistics_active_completed_tasks, ACTIVE_COMPLETED_TASKS);
```

Comment:
This seems to be unneeded. If you would remove this line, the test would work as well. 
You are testing if `ResourceProvider` was called, not it the value was used